equinix/resource_metal_connection.go

```diff
@@ -104,8 +104,8 @@ func resourceMetalConnection() *schema.Resource {
 			},
 			"project_id": {
 				Type:        schema.TypeString,
-				Required:    true,
-				Description: "ID of the project where the connection is scoped to, only used for type == \"shared\"",
+				Optional:    true,
+				Description: "ID of the project where the connection is scoped to. Required with type \"shared\"",
```

Comment:
I don't know which of these is correct. The previous hunk that reverts `ret[order[p.Role]] = connPort` has me calling everything in this patch into question.

Reply:
Thanks for the detailed review @displague, I will check again all the changes. In this case, both project and organization should be optional. Some of the introduced errors you mentioned were my fault in choosing the wrong option when reviewing the rejects.